core/src/main/java/org/keycloak/jose/jwk/JWKParser.java

```diff
@@ -48,4 +60,65 @@ public JWKParser parse(String jwk) {
         }
     }
 
+    @Override
+    public PublicKey toPublicKey() {
+        String keyType = jwk.getKeyType();
```

Comment:
I think there is way to create this without jwk (see `static JWKParser create()` above) so maybe better reverse the following checks to not have a NullPointerException.

Reply:
Good point.

I've just pretty much copy/pasted this part from the original `JWKParser` class in the `java16` directory as there is no need to have specific code for java16 or bigger. But at the same time, it doesn't hurt to improve the code and make it more robust when on this :-) I did the same for `AbstractJWKParser` as well. Also made `ServerJWKParser` to not be abstract as I think it was `abstract` by accident and `abstract` caused this test to not be executed.